packages/network-controller/src/NetworkController.ts

```diff
@@ -310,26 +302,14 @@ export class NetworkController extends BaseControllerV2<
     this.state.network === 'loading' && this.lookupNetwork();
   }
 
-  /**
-   * Sets a new configuration for web3-provider-engine.
-   *
-   * TODO: Replace this wth a method.
-   *
-   * @param providerConfig - The web3-provider-engine configuration.
-   */
-  set providerConfig(providerConfig: ProviderConfig) {
-    this.internalProviderConfig = providerConfig;
```

Comment:
This "internal provider config" is critically important for mobile today because they use it to enable sending transactions. However I do have [a PR up on mobile](https://github.com/MetaMask/metamask-mobile/pull/5619) that removes the use of that configuration.

Not sure what's best to do here :thinking: We could preserve it for now, and remove it in a later step. But that would add another step to the normalization effort. Whereas if we remove it here, we'll have to note this breaking change in the changelog and sequence this update to come _after_ that related mobile PR.

Probably best to keep this as-is, and sequence this update to come after we remove the need for the config.

Reply:
@Gudahtt Ah, I advised that @cryptodev-2s remove `internalProviderConfig`, but didn't realize it was still being used.

This begs a question. Previously, when `providerConfig` was set, `internalProviderConfig` was set to the same provider object. Now that `initializeProvider` no longer takes a provider config object, but relies solely on state, it seems like we could take one of two paths here:

A. Ensure `internalProviderConfig` reflects the provider config in state each time `initializeProvider` is called. (This assumes that `initializeProvider` may get called multiple times.)
B. Set `internalProviderConfig` once in the constructor to the provider config object passed to the constructor, but do not attempt to synchronize it after that. (This assumes that `initializeProvider` will only get called once.)

Which path is better?